app/browser/api/ledger.js

```diff
@@ -2457,6 +2458,117 @@ const transitionWalletToBat = () => {
   }
 }
 
+let currentMediaKey = null
+const onMediaRequest = (state, xhr, type, tabId) => {
+  if (!xhr || type == null) {
+    return state
+  }
+
+  const parsed = ledgerUtil.getMediaData(xhr, type)
+  const mediaId = ledgerUtil.getMediaId(parsed, type)
+  const mediaKey = ledgerUtil.getMediaKey(mediaId, type)
+  let duration = ledgerUtil.getMediaDuration(parsed, type)
+
+  if (mediaId == null || duration == null || mediaKey == null) {
+    return state
+  }
+
+  const minDuration = getSetting(settings.PAYMENTS_MINIMUM_VISIT_TIME)
+  duration = parseInt(duration)
+  if (duration > 0 && duration < minDuration) {
+    duration = minDuration
+  }
+
+  if (!ledgerPublisher) {
+    ledgerPublisher = require('bat-publisher')
+  }
+
+  let revisited = true
+  const activeTabId = tabState.getActiveTabId(state)
+  if (activeTabId === tabId && mediaKey !== currentMediaKey) {
+    revisited = false
+    currentMediaKey = mediaKey
+  }
+
+  const cache = ledgerVideoCache.getDataByVideoId(state, mediaKey)
+
+  if (!cache.isEmpty()) {
+    return module.exports.saveVisit(state, cache.get('publisher'), duration, revisited)
+  }
+
+  const options = underscore.extend({roundtrip: roundtrip}, clientOptions)
+  const mediaProps = {
+    mediaId,
+    providerName: type
+  }
+
+  ledgerPublisher.getMedia.getPublisherFromMediaProps(mediaProps, options, (error, response) => {
```

Comment:
1. bat-publisher should throw an error if the media URL protocol is not "https:"
2. it appears `bat-publisher` is making a call to a youtube.com page from the browser main process, scraping and parsing the page HTML, and extracting page data from it. from a security perspective, it would be much better to do this in a page content script that runs when the user visits a youtube page (ex: how we currently extract favicons from publisher pages).

Reply:
cc @mrose17

Reply:
1. `bat-publisher` is being passed a URL that the browser is playing. whether it is `http?` is immaterial. the goal is to look at the URL string and figure out where to look for the oembed data.

2. i think that this is a worthy future, but not essential at this point. it is certainly good to have routines that parse data run in a separate context. it's not clear to me why that is "more necessary" in this case. we can open up an issue for this, and when we do, it would be worthwhile to see what other cases might exist!

Reply:
@mrose17 I think comment 1 is still unaddressed? I'd also strongly prefer we address 2 before moving on. The follow up issue seems to be to address the general case, but no reason for us to not adhere to the correct practices now.

Reply:
@mrose17 I'm going to hold off on merge until @evq 's concern can be addressed (and we have a sign off in https://github.com/brave/internal/issues/141). Let us know if we need to update libs after that is addressed

Reply:
1. is addressed. we aren't opening a URL, we are parsing it in order to find an oembed provider. if the browser decides to play it and then ask for information, then the module will provide the information.  however, in the interest of moving things along, i have pushed `bat-publisher@1.2.3` that makes this change.

2. it is not a simple thing to make this change (it has to be made in the browser). nor am i convinced that it makes sense to require this code be held to a standard higher than any other code in the browser. as i noted in my comment.

everything has a cost. i suppose we can address 1, because it's a small change, even if i do view it as unnecessary. in addressing 2, it has a large cost (i already researched it with @NejcZdovc) and addresses a theoretical concern. there is significant cost to that we will not be able to merge this upcoming week. i don't have an issue with the browser folks reviewing the issue that @diracdeltas created after this merge, and having the same standard applied to the entire browser. i most certainly have an issue having this new standard being applied in this context and in this timeline.